hknweb/academics/models/course_surveys.py

```diff
@@ -1,17 +1,21 @@
 from django.db import models
 from django.core.validators import MinValueValidator
+from django.db.models import Case, When, Value
 
 from .base_models import AcademicEntity
+from .icsr import _chrono
 
 
 class Question(AcademicEntity):
-    pass
+    recent_semester = models.ForeignKey("Semester", on_delete=models.PROTECT, blank=True, null=True)

```

Comment:
Why do we need a `recent_semester` on `Question`? I presume it's in part to help in calculating the `current_text` attr; I'm kind of just curious as to what your thoughts on this are!

Reply:
Hey! You're pretty much right that it's just to calculate current_text; recent_semester keeps track of the last semester in which the question was asked. I felt that it might also be useful information otherwise to see the most recent questions very easily, so I made it a model field.